baiji/s3.py

```diff
@@ -880,27 +894,56 @@ def _get_etag(self, netloc, remote_path):
             raise KeyNotFound("s3://%s/%s not found on s3" % (netloc, remote_path))
         return k.etag.strip("\"") # because s3 seriously gives the md5sum back wrapped in an extra set of double quotes...
 
+    def _build_etag(self, local_path, n_parts, part_size):
```

Comment:
Can you add a high-level description of what you're doing here, with any relevant links you have on hand? For someone lacking context, the algorithm will seem a bit mysterious.